### ISSUE TITLE
OpenJ9 - JVM crash with GPF EVENT

### ISSUE THREAD

**Issue by EilonSt** [eclipse-openj9/openj9#16341]
**### Java - version output:**
openjdk version "11.0.14.1" 2022-02-08
IBM Semeru Runtime Open Edition 11.0.14.1 (build 11.0.14.1+1)
Eclipse OpenJ9 VM 11.0.14.1 (build openj9-0.30.1, JRE 11 Linux amd64-64-Bit Compressed References 20220210_379 (JIT enabled, AOT enabled)
OpenJ9   - 9dccbe076
OMR      - 56c3376ba
JCL      - 5c423dacd0 based on jdk-11.0.14.1+1)

**### Summary of problem:**
Every ~ 1.5 hours , I got a JVM crash, without real understanding what causing the crash. The crash is generated by a GPF. See below some extract from the generated javacore file:

```
0SECTION       TITLE subcomponent dump routine
NULL           ===============================
1TICHARSET     UTF-8
1TISIGINFO     Dump Event "gpf" (00002000) received
1TIDATETIMEUTC Date: 2022/11/18 at 20:16:20:536 (UTC)
1TIDATETIME    Date: 2022/11/18 at 15:16:20:536
1TITIMEZONE    Timezone: UTC-5 (EST)
1TINANOTIME    System nanotime: 115990214800104
1TIFILENAME    Javacore filename:    /tmp/javacore.20221118.151602.2783.0002.txt
1TIREQFLAGS    Request Flags: 0x41 (exclusive+preempt)
1TIPREPSTATE   Prep State: 0x80 (trace_disabled)
1TIPREPINFO    Exclusive VM access not taken: data may not be consistent across javacore sections

```

The thread that was running at the crash:

```
NULL           ------------------------------------------------------------------------
0SECTION       THREADS subcomponent dump routine
NULL           =================================
NULL
1XMPOOLINFO    JVM Thread pool info:
2XMPOOLTOTAL       Current total number of pooled threads: 77
2XMPOOLLIVE        Current total number of live threads: 61
2XMPOOLDAEMON      Current total number of live daemon threads: 60
NULL
1XMCURTHDINFO  Current thread
3XMTHREADINFO      "JIT Compilation Thread-000" J9VMThread:0x000000000001E600, omrthread_t:0x00007F86D01436E0, java/lang/Thread:0x00000000A00FA378, state:R, prio=10
3XMJAVALTHREAD            (java/lang/Thread getId:0x3, isDaemon:true)
3XMTHREADINFO1            (native thread ID:0xAE3, native priority:0xB, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00000000)
3XMTHREADINFO2            (native stack address range from:0x00007F86AF03B000, to:0x00007F86AF13B000, size:0x100000)
3XMCPUTIME               CPU usage total: 9.279891032 secs, current category="JIT"
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           No Java callstack associated with this thread
3XMTHREADINFO3           Native callstack:
4XENATIVESTACK               (0x00007F86D6A87542 [libj9prt29.so+0x26542])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D6A875BE [libj9prt29.so+0x265be])
4XENATIVESTACK               (0x00007F86D6A876B4 [libj9prt29.so+0x266b4])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D6A8741B [libj9prt29.so+0x2641b])
4XENATIVESTACK               (0x00007F86D6A88412 [libj9prt29.so+0x27412])
4XENATIVESTACK               (0x00007F86D6A891B4 [libj9prt29.so+0x281b4])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D61FD626 [libj9dmp29.so+0x1b626])
4XENATIVESTACK               (0x00007F86D61FD7BD [libj9dmp29.so+0x1b7bd])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D61F9DCD [libj9dmp29.so+0x17dcd])
4XENATIVESTACK               (0x00007F86D61F4E2D [libj9dmp29.so+0x12e2d])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D61F620F [libj9dmp29.so+0x1420f])
4XENATIVESTACK               (0x00007F86D62000FC [libj9dmp29.so+0x1e0fc])
4XENATIVESTACK               (0x00007F86D61E6D9D [libj9dmp29.so+0x4d9d])
4XENATIVESTACK               (0x00007F86D61E6475 [libj9dmp29.so+0x4475])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D61E9AEB [libj9dmp29.so+0x7aeb])
4XENATIVESTACK               (0x00007F86D61E9C7F [libj9dmp29.so+0x7c7f])
4XENATIVESTACK               (0x00007F86D620254E [libj9dmp29.so+0x2054e])
4XENATIVESTACK               (0x00007F86D6D2D4D2 [libj9vm29.so+0x3f4d2])
4XENATIVESTACK               (0x00007F86D6A8B893 [libj9prt29.so+0x2a893])
4XENATIVESTACK               (0x00007F86D6D2D6E5 [libj9vm29.so+0x3f6e5])
4XENATIVESTACK               (0x00007F86D6A8AD5A [libj9prt29.so+0x29d5a])
4XENATIVESTACK               (0x00007F86D9176630 [libpthread.so.0+0xf630])
4XENATIVESTACK               raise+0x2b (0x00007F86D91764FB [libpthread.so.0+0xf4fb])
4XENATIVESTACK               (0x00007F86D54F6797 [libj9jit29.so+0x56d797])
4XENATIVESTACK               (0x00007F86D54F6822 [libj9jit29.so+0x56d822])
4XENATIVESTACK               (0x00007F86D52F760B [libj9jit29.so+0x36e60b])
4XENATIVESTACK               (0x00007F86D559C887 [libj9jit29.so+0x613887])
4XENATIVESTACK               (0x00007F86D52DDC31 [libj9jit29.so+0x354c31])
4XENATIVESTACK               (0x00007F86D55A41DE [libj9jit29.so+0x61b1de])
4XENATIVESTACK               (0x00007F86D52D4846 [libj9jit29.so+0x34b846])
4XENATIVESTACK               (0x00007F86D56B27C7 [libj9jit29.so+0x7297c7])
4XENATIVESTACK               (0x00007F86D56B2D49 [libj9jit29.so+0x729d49])
4XENATIVESTACK               (0x00007F86D56B410B [libj9jit29.so+0x72b10b])
4XENATIVESTACK               0x00007F86D54B6165
4XENATIVESTACK               0x00007F86D50D831B
4XENATIVESTACK               0x00007F86D50D9357
4XENATIVESTACK               0x00007F86D6A8B893
4XENATIVESTACK               0x00007F86D50D6AF9
4XENATIVESTACK               0x00007F86D50D7120
4XENATIVESTACK               0x00007F86D50D5C73
4XENATIVESTACK               0x00007F86D50D6152
4XENATIVESTACK               0x00007F86D50D61FF
4XENATIVESTACK               0x00007F86D6A8B893
4XENATIVESTACK               0x00007F86D50D6632
NULL
```

**Comment by DanHeidinga** [comment]
@0xdaryl Can you get someone on the jit team to look at this?  Seems like a crash in the jit based on the stack trace of the current thread.

@EilonSt Can you share the system core, javacore, and any other diagnostic files from the crash?

**Comment by EilonSt** [comment]
Hi @DanHeidinga thanks for the quick response.
adding zip file with all the relevant files for diagnosing the crash.
[jvm-crash-diagnostics-files.zip](https://github.com/eclipse-openj9/openj9/files/10050477/jvm-crash-diagnostics-files.zip)

regarding the system core file it is too big for here so let me know if you need it and if so where can I send it to you.
Thanks.

**Comment by pshipton** [comment]
@EilonSt you can use any file sharing service to transfer the core. It may also be worth trying the latest release 11.0.17.

The javacore contains the following information.
```
1XHEXCPMODULE  Compiling method: org/apache/activemq/store/kahadb/MessageDatabase.store(Lorg/apache/activemq/store/kahadb/JournalCommand;ZLorg/apache/activemq/store/kahadb/MessageDatabase$IndexAware;Ljava/lang/Runnable;Ljava/lang/Runnable;)Lorg/apache/activemq/store/kahadb/disk/journal/Location;
NULL
1XHFLAGS       VM flags:00000000000502FF
```

`vmState [0x502ff]: {J9VMSTATE_JIT} {targetedInlining}`

**Comment by 0xdaryl** [comment]
@jdmpapin : please investigate this crash in inlining when the core file is made available.

**Comment by EilonSt** [comment]
please see the relevant files for diagnosing the crash including the system core here - https://drive.google.com/drive/folders/1vtnCT33oAXPkSht6scM7RSgr64NhrZk2?usp=sharing

**Comment by jdmpapin** [comment]
I'm pretty sure this is a dup of #14526, which was fixed in #14886 (and #14898), so the fix should be in any newer release, i.e. 11.0.15 and later. @EilonSt, can you please try a newer release?

**Comment by EilonSt** [comment]
Hi @jdmpapin I tried to use release 11.0.17 and it seems that the problem solved. thank you every one for the help!
